packages/components/src/components/checkbox/checkbox.jsx

```diff
@@ -12,33 +12,54 @@ class Checkbox extends React.PureComponent {
         };
```

Comment:
Since we are syncing with `props.defaultChecked` in `getDerivedStateFromProps` I don't think this is necessary anymore.

Reply:
ahh okay, let me check

Reply:
@reza-fs it has its own internal `checked` state, which is toggled whenever we click on the checkbox. we need this function to toggle checkbox value from outside